### ISSUE TITLE
Not working

### ISSUE THREAD

**Issue by ketakachono** [chaofengc/Face-SPARNet#28]
(sparnet) C:\Users\user\sparnet>python test_enhance_single_unalign.py --gpus 1 --model sparnethd --name SPARNetHD_V4_Attn2D --res_depth 10 --att_name spar --Gnorm "in" --pretrain_model_path pretrain_models/SPARNetHD_V4_Attn2D_net_H-epoch10.pth --test_img_path ./test_images/test_hzgg.jpg --results_dir test_hzgg_results
----------------- Options ---------------
                    Dnorm: none
                    Gnorm: in                                   [default: bn]
                 att_name: spar
               batch_size: 32
          bottleneck_size: 4
          checkpoints_dir: ./check_points
              data_device: cuda:0                               [default: None]
                 dataroot: None
             dataset_name: celeba
                    debug: False
                   device: cuda:0                               [default: None]
                    epoch: latest
                     eval: False
                  gpu_ids: [0]                                  [default: None]
                     gpus: 1
                init_gain: 0.02
                init_type: normal
                 input_nc: 3
                  isTrain: False                                [default: None]
                load_iter: 0                                    [default: 0]
                load_size: None
         max_dataset_size: inf
                    model: sparnethd                            [default: test]
               n_layers_D: 4
                     name: SPARNetHD_V4_Attn2D                  [default: experiment_name]
                      ndf: 64
                      ngf: 64
                    ntest: inf
                    num_D: 3
                 num_test: 50
              num_threads: 8
                output_nc: 3
                    phase: test
               preprocess: none
      pretrain_model_path: pretrain_models/SPARNetHD_V4_Attn2D_net_H-epoch10.pth        [default: ]
                res_depth: 10
              results_dir: test_hzgg_results                    [default: ./results/]
              save_as_dir:
                     seed: 123
           serial_batches: False
                   suffix:
            test_img_path: ./test_images/test_hzgg.jpg          [default: ]
             test_upscale: 1
                  verbose: False
----------------- End -------------------
======> Loading images, crop and align faces.
======> Saving aligned LQ faces to test_hzgg_results\LQ_faces
initialize network SPARNet with normal
model [SPARNetHDModel] was created
Loading pretrained model pretrain_models/SPARNetHD_V4_Attn2D_net_H-epoch10.pth
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [08:16<00:00, 496.16s/it]
======> Save the enhanced faces.
======> Paste the enhanced faces back to the original image.
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  8.58it/s]
======> Save final result to test_hzgg_results\hq_final.jpg

(sparnet) C:\Users\user\sparnet>

![hq_final](https://github.com/chaofengc/Face-SPARNet/assets/130495779/8426ca57-a778-4917-a999-5a1894043ad5)

**Comment by ketakachono** [comment]
(sparnet) C:\Users\user\sparnet>conda list
# packages in environment at C:\Users\user\miniconda3\envs\sparnet:
#
# Name                    Version                   Build  Channel
absl-py                   1.4.0                    pypi_0    pypi
astor                     0.8.1                    pypi_0    pypi
blas                      1.0                         mkl
ca-certificates           2023.5.7             h56e8100_0    conda-forge
certifi                   2023.5.7           pyhd8ed1ab_0    conda-forge
cmake                     3.26.3                   pypi_0    pypi
colorama                  0.4.6                    pypi_0    pypi
cudatoolkit               10.1.243             h74a9793_0
cycler                    0.11.0                   pypi_0    pypi
dlib                      19.24.0          py37h53cf009_0    conda-forge
fonttools                 4.38.0                   pypi_0    pypi
freetype                  2.12.1               ha860e81_0
future                    0.18.3                   pypi_0    pypi
gast                      0.2.2                    pypi_0    pypi
giflib                    5.2.1                h8cc25b3_3
google-pasta              0.2.0                    pypi_0    pypi
grpcio                    1.54.2                   pypi_0    pypi
h5py                      3.8.0                    pypi_0    pypi
imageio                   2.9.0                    pypi_0    pypi
imgaug                    0.4.0                    pypi_0    pypi
importlib-metadata        6.6.0                    pypi_0    pypi
intel-openmp              2021.4.0          haa95532_3556
jpeg                      9e                   h2bbff1b_1
keras-applications        1.0.8                    pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.4.4                    pypi_0    pypi
lerc                      3.0                  hd77b12b_0
libdeflate                1.17                 h2bbff1b_0
libpng                    1.6.39               h8cc25b3_0
libtiff                   4.5.0                h6c2663c_2
libwebp                   1.2.4                hbc33d0d_1
libwebp-base              1.2.4                h2bbff1b_1
lz4-c                     1.9.4                h2bbff1b_0
markdown                  3.4.3                    pypi_0    pypi
markupsafe                2.1.2                    pypi_0    pypi
matplotlib                3.5.3                    pypi_0    pypi
mkl                       2021.4.0           haa95532_640
mkl-service               2.4.0            py37h2bbff1b_0
mkl_fft                   1.3.1            py37h277e83a_0
mkl_random                1.2.2            py37hf11a4ad_0
networkx                  2.6.3                    pypi_0    pypi
ninja                     1.10.2               haa95532_5
ninja-base                1.10.2               h6d14046_5
numpy                     1.21.5           py37h7a0a035_3
numpy-base                1.21.5           py37hca35cd5_3
opencv-python             4.7.0.72                 pypi_0    pypi
openssl                   1.1.1t               h2bbff1b_0
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 23.1                     pypi_0    pypi
pillow                    7.2.0                    pypi_0    pypi
pip                       22.3.1           py37haa95532_0
protobuf                  4.23.0                   pypi_0    pypi
pyparsing                 3.0.9                    pypi_0    pypi
python                    3.7.16               h6244533_0
python-dateutil           2.8.2                    pypi_0    pypi
python_abi                3.7                     2_cp37m    conda-forge
pytorch                   1.5.1           py3.7_cuda101_cudnn7_0    pytorch
pytorch-fid               0.3.0                    pypi_0    pypi
pywavelets                1.3.0                    pypi_0    pypi
scikit-image              0.17.2                   pypi_0    pypi
scipy                     1.4.1                    pypi_0    pypi
setuptools                65.6.3           py37haa95532_0
shapely                   2.0.1                    pypi_0    pypi
six                       1.16.0             pyhd3eb1b0_1
sqlite                    3.41.2               h2bbff1b_0
tensorboard               1.15.0                   pypi_0    pypi
tensorboardx              2.1                      pypi_0    pypi
tensorflow                1.15.0                   pypi_0    pypi
tensorflow-estimator      1.15.1                   pypi_0    pypi
termcolor                 2.3.0                    pypi_0    pypi
tifffile                  2021.11.2                pypi_0    pypi
tk                        8.6.12               h2bbff1b_0
torchvision               0.6.1                py37_cu101    pytorch
tqdm                      4.65.0                   pypi_0    pypi
typing-extensions         4.5.0                    pypi_0    pypi
vc                        14.2                 h21ff451_1
vs2015_runtime            14.27.29016          h5e58377_2
werkzeug                  2.2.3                    pypi_0    pypi
wheel                     0.38.4           py37haa95532_0
wincertstore              0.2              py37haa95532_2
wrapt                     1.15.0                   pypi_0    pypi
xz                        5.4.2                h8cc25b3_0
zipp                      3.15.0                   pypi_0    pypi
zlib                      1.2.13               h8cc25b3_0
zstd                      1.5.5                hd43e919_0

(sparnet) C:\Users\user\sparnet>

**Comment by ketakachono** [comment]
python test_enhance_single_unalign.py --gpus 1 --model sparnethd --name SPARNetHD_V4_Attn3D --res_depth 10 --att_name spar3d --Gnorm "in" --pretrain_model_path ./pretrain_models/SPARNetHD_V4_Attn3D_net_H-epoch10.pth --test_img_path ./test_images/test_hzgg.jpg --results_dir test_hzgg_results

![000](https://github.com/chaofengc/Face-SPARNet/assets/130495779/d77713f2-9ab8-468f-9473-f4f9139b25f8)
